src/test/java/uk/gov/pay/card/resources/HealthCheckResourceTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.pay.card.resources;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.jayway.jsonassert.JsonAssert;
+import io.dropwizard.setup.Environment;
+import org.hamcrest.core.Is;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.ws.rs.core.Response;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class HealthCheckResourceTest {
+    @Mock
+    private Environment environment;
+
+    @Mock
+    private HealthCheckRegistry healthCheckRegistry;
+
+    private HealthCheckResource resource;
+
+    @Before
+    public void setup() {
+        when(environment.healthChecks()).thenReturn(healthCheckRegistry);
+        resource = new HealthCheckResource(environment);
+    }
+
+    @Test
+    public void checkHealthCheck_isUnHealthy() throws JsonProcessingException {
+        SortedMap<String,HealthCheck.Result> map = new TreeMap<>();
+        map.put("ping", HealthCheck.Result.unhealthy("application is unavailable"));
+        map.put("deadlocks", HealthCheck.Result.unhealthy("no new threads available"));
+        when(healthCheckRegistry.runHealthChecks()).thenReturn(map);
+        Response response = resource.healthCheck();
+        assertThat(response.getStatus(), is(503));
+        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        String body = ow.writeValueAsString(response.getEntity());
+
+        JsonAssert.with(body)
+                .assertThat("$.*", hasSize(2))
+                .assertThat("$.ping.healthy", Is.is(false))
+                .assertThat("$.deadlocks.healthy", Is.is(false));
+    }
```

Comment:
Judicious use of spaces can help clarity in tests like this:

```java
SortedMap<String,HealthCheck.Result> map = new TreeMap<>();
map.put("ping", HealthCheck.Result.unhealthy("application is unavailable"));
map.put("deadlocks", HealthCheck.Result.unhealthy("no new threads available"));

when(healthCheckRegistry.runHealthChecks()).thenReturn(map);

Response response = resource.healthCheck();

assertThat(response.getStatus(), is(503));

ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
String body = ow.writeValueAsString(response.getEntity());
JsonAssert.with(body)
        .assertThat("$.*", hasSize(2))
        .assertThat("$.ping.healthy", is(false))
        .assertThat("$.deadlocks.healthy", is(false));
```

My rule of thumb is to have blank lines between the stubbing and the actual method call under test, and between the actual method call and the assertions and verifications, and anywhere else it makes sense. So it’s easy to see at a glance, “here’s the setup, here’s the actual thing we want to test, here’s the checks that the actual thing worked”.

Reply:
Consider the messenger shot :) All I did was copy this across from publicapi.. I'll fix up here and raise a PR against publicapi too